diff --git a/node_modules/@newrelic/gatsby-theme-newrelic/gatsby-node.js b/node_modules/@newrelic/gatsby-theme-newrelic/gatsby-node.js
index 4641d4b..c4c6150 100644
--- a/node_modules/@newrelic/gatsby-theme-newrelic/gatsby-node.js
+++ b/node_modules/@newrelic/gatsby-theme-newrelic/gatsby-node.js
@@ -288,7 +288,7 @@ exports.onCreatePage = (helpers, themeOptions) => {
       ) {
         createPage({
           ...transformedPage,
-          path: path.join(
+          path: path.posix.join(
             `/${locale}`,
             transformedPage.path,
             forceTrailingSlashes ? '/' : ''
diff --git a/node_modules/@newrelic/gatsby-theme-newrelic/gatsby/constants.js b/node_modules/@newrelic/gatsby-theme-newrelic/gatsby/constants.js
index 9358633..d7178ea 100644
--- a/node_modules/@newrelic/gatsby-theme-newrelic/gatsby/constants.js
+++ b/node_modules/@newrelic/gatsby-theme-newrelic/gatsby/constants.js
@@ -6,4 +6,4 @@ const TESSEN_PATH = path.resolve(
   `../static/tessen.min-${TESSEN_VERSION}.js`
 );
 
-module.exports = { TESSEN_PATH };
+module.exports = { TESSEN_PATH: TESSEN_PATH };
