diff --git a/node_modules/gatsby/cache-dir/find-path.js b/node_modules/gatsby/cache-dir/find-path.js
index 5b13ad9..85bdbdc 100644
--- a/node_modules/gatsby/cache-dir/find-path.js
+++ b/node_modules/gatsby/cache-dir/find-path.js
@@ -1,6 +1,7 @@
 import { pick } from "@reach/router/lib/utils"
 import stripPrefix from "./strip-prefix"
 import normalizePagePath from "./normalize-page-path"
+import { maybeGetBrowserRedirect } from "./redirects.js"
 
 const pathCache = new Map()
 let matchPaths = []
@@ -112,6 +113,11 @@ export const findPath = rawPathname => {
     return pathCache.get(trimmedPathname)
   }
 
+  const redirect = maybeGetBrowserRedirect(rawPathname)
+  if (redirect) {
+    return findPath(redirect.toPath)
+  }
+
   let foundPath = findMatchPath(trimmedPathname)
 
   if (!foundPath) {
diff --git a/node_modules/gatsby/cache-dir/navigation.js b/node_modules/gatsby/cache-dir/navigation.js
index 62e1832..39f461b 100644
--- a/node_modules/gatsby/cache-dir/navigation.js
+++ b/node_modules/gatsby/cache-dir/navigation.js
@@ -1,7 +1,7 @@
 import React from "react"
 import PropTypes from "prop-types"
 import loader, { PageResourceStatus } from "./loader"
-import redirects from "./redirects.json"
+import { maybeGetBrowserRedirect } from "./redirects.js"
 import { apiRunner } from "./api-runner-browser"
 import emitter from "./emitter"
 import { RouteAnnouncerProps } from "./route-announcer-props"
@@ -9,24 +9,8 @@ import { navigate as reachNavigate } from "@reach/router"
 import { globalHistory } from "@reach/router/lib/history"
 import { parsePath } from "gatsby-link"
 
-// Convert to a map for faster lookup in maybeRedirect()
-
-const redirectMap = new Map()
-const redirectIgnoreCaseMap = new Map()
-
-redirects.forEach(redirect => {
-  if (redirect.ignoreCase) {
-    redirectIgnoreCaseMap.set(redirect.fromPath, redirect)
-  } else {
-    redirectMap.set(redirect.fromPath, redirect)
-  }
-})
-
 function maybeRedirect(pathname) {
-  let redirect = redirectMap.get(pathname)
-  if (!redirect) {
-    redirect = redirectIgnoreCaseMap.get(pathname.toLowerCase())
-  }
+  const redirect = maybeGetBrowserRedirect(pathname)
 
   if (redirect != null) {
     if (process.env.NODE_ENV !== `production`) {
@@ -72,10 +56,7 @@ const navigate = (to, options = {}) => {
   }
 
   let { pathname } = parsePath(to)
-  let redirect = redirectMap.get(pathname)
-  if (!redirect) {
-    redirect = redirectIgnoreCaseMap.get(pathname.toLowerCase())
-  }
+  const redirect = maybeGetBrowserRedirect(pathname)
 
   // If we're redirecting, just replace the passed in pathname
   // to the one we want to redirect to.
diff --git a/node_modules/gatsby/cache-dir/redirects.js b/node_modules/gatsby/cache-dir/redirects.js
new file mode 100644
index 0000000..0f58c6d
--- /dev/null
+++ b/node_modules/gatsby/cache-dir/redirects.js
@@ -0,0 +1,22 @@
+import redirects from "./redirects.json"
+
+// Convert to a map for faster lookup in maybeRedirect()
+
+const redirectMap = new Map()
+const redirectIgnoreCaseMap = new Map()
+
+redirects.forEach(redirect => {
+  if (redirect.ignoreCase) {
+    redirectIgnoreCaseMap.set(redirect.fromPath, redirect)
+  } else {
+    redirectMap.set(redirect.fromPath, redirect)
+  }
+})
+
+export function maybeGetBrowserRedirect(pathname) {
+  let redirect = redirectMap.get(pathname)
+  if (!redirect) {
+    redirect = redirectIgnoreCaseMap.get(pathname.toLowerCase())
+  }
+  return redirect
+}
